Lib/test/support/__init__.py

```diff
@@ -1982,3 +1983,13 @@ def skip_if_broken_multiprocessing_synchronize():
             synchronize.Lock(ctx=None)
         except OSError as exc:
             raise unittest.SkipTest(f"broken multiprocessing SemLock: {exc!r}")
+
+
+def check_disallow_instantiation(testcase, tp, *args, **kwds):
+    """
+    Helper for testing types with the Py_TPFLAGS_DISALLOW_INSTANTIATION flag.
```

Comment:
I suggest to document the behaviour ("Check that a type cannot be instanciated") rather than mentioning the C API, but you can mention Py_TPFLAGS_DISALLOW_INSTANTIATION ;-)

The test suite is used on other Python implementations, like PyPy, not only CPython.

Reply:
I'll address your review comments tomorrow. Thanks for reviewing :)